src/orders/repository/order.repository.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Order } from '../domain/order.domain';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class OrderRepository {
+  constructor(
+    @InjectRepository(Order)
+    private readonly orderRepository: Repository<Order>,
+  ) {}
+
+  async getAll(): Promise<Order[]> {
+    return this.orderRepository.find();
+  }
+
+  async getById(id: string): Promise<Order | null> {
+    return this.orderRepository.findOneBy({ id });
+  }
+
+  async create(order: Order): Promise<Order> {
+    return await this.orderRepository.save(order);
+  }
+
+  async update(id: string, newOrder: Order): Promise<Order> {
```

Comment:
Here think what happens if the order does not exists. It will be inserted and not updated. You need some additional if statement.